bigram_builder.py

```diff
@@ -0,0 +1,12 @@
+from os import path
+
+from pkg.indexbuilder import BigramIndexBuilder
+from pkg.context import Context
+
+corpus_path = "data/corpus/UofO_Courses.yaml"
+dictionary_path = "data/dictionary/UofOCourses.txt"
+index_path = path.realpath("data/index/UofO_Courses.yaml")
+
+ctx = Context(corpus_path, dictionary_path, index_path)
+
+BigramIndexBuilder(ctx).build()
```

Comment:
Have you given much thought about when/how to load all these indices? Note that loading the primary index here necessarily involves loading the secondary index. I'm thinking we should do all this heavy work right when we start the program, so we don't add any lag during actual operation. It's also a good place to ensure everything is loaded according to the execution context (filters, normalizers, etc.).

Another thought I had is that we should hash different context configs and append that SHA to the documents it generates. That way we can keep different versions of the dict/indices, etc. in the repo without having to ever regenerate if we change the config. what do you think?

Reply:
I agree, we could have some initialization function to load everything up. I think it looks bad when we load everything after hitting search, since it's hard to provide feedback to the user so its evident something is happening. 
So with the SHA, we would basically end up with all the dict/index "options" stored in the repo, so we wouldn't have to rebuild them each time, but just find the appropriate file? I'm not super familiar with SHA stuff / what the cost is to hash something, but we could probably also just append the config settings plainly and check for a match that way if its less complicated? Either way, I think it sounds solid.

Reply:
The cost to hash something is constant time, so there's no worry there.
I'll prototype my idea and throw up a PR, hopefully tonight.